### PR TITLE
Added espresso tests for sign-up flow. No robots or pages.

### PR DESCRIPTION
**Pull request by matthew-carroll**

### CODE REVIEW

File: app/src/androidTest/java/io/mattcarroll/androidtesting/EspressoSignUpTest.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+        // Ensure that this Activity destroyed itself upon successful completion of sign up.
+        assertTrue(activityRule.getActivity().isDestroyed());
+    }
+
+    @Test
+    public void userSignUpPersonalInfoVerifyRequiredFieldsAreRequired() {
+        // Verify required fields show errors and non-required fields do not.
+        onView(withId(R.id.button_next))
+                .perform(scrollTo())
```
Comment (bamatson) [inline app/src/androidTest/java/io/mattcarroll/androidtesting/EspressoSignUpTest.java]: FYI you can perform multiple ViewActions here, i.e. `perform(scrollTo(), click())`

File: app/src/androidTest/java/io/mattcarroll/androidtesting/EspressoSignUpTest.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        // Choose credentials and sign up.
+        enterCredentials("myuser", "123456");
+        scrollToAndPressNext();
+
+        // Ensure that this Activity destroyed itself upon successful completion of sign up.
+        assertTrue(activityRule.getActivity().isDestroyed());
+    }
+
+    @Test
+    public void userSignUpPersonalInfoVerifyRequiredFieldsAreRequired() {
```
Comment (bamatson) [inline app/src/androidTest/java/io/mattcarroll/androidtesting/EspressoSignUpTest.java]: Should this be more granular, with a test per field?

File: app/src/androidTest/java/io/mattcarroll/androidtesting/EspressoSignUpTest.java (hunk 1 above)
Comment (matthew-carroll) [inline app/src/androidTest/java/io/mattcarroll/androidtesting/EspressoSignUpTest.java]: done

File: app/src/androidTest/java/io/mattcarroll/androidtesting/EspressoSignUpTest.java (hunk 2 above)
Comment (matthew-carroll) [inline app/src/androidTest/java/io/mattcarroll/androidtesting/EspressoSignUpTest.java]: probably not...there's a time penalty with functional tests to relaunch the Activity and redo any steps leading up to this point...so I dunno if its worth it with espresso tests...

**Pull request merged**